.github/workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: CI Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build_lint_test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Setup Mage
+        uses: magefile/mage-action@v3
+        with:
+          install-only: true
+
+      - name: Install test dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y gcc
+
+      - name: Setup CI Linter
+        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.54.2
```

Comment:
We are using `golangci-lint` version 1.54.2, the latest is 1.61.0. If we use 1.61.0 followings are need to be fixed.
```
data/frame_json.go:557:16: G115: integer overflow conversion int64 -> int32 (gosec)
			return int32(iV), err
			            ^
data/time_series.go:69:5: S1009: should omit nil check; len() for []*github.com/grafana/grafana-plugin-sdk-go/data.Field is defined as zero (gosimple)
	if f.Fields == nil || len(f.Fields) == 0 {
	   ^
data/time_series_field_sort_test.go:122:12: printf: non-constant format string in call to (*testing.common).Logf (govet)
				t.Logf(tt.frameToSort.StringTable(-1, -1))
				       ^
data/time_series_field_sort_test.go:213:12: printf: non-constant format string in call to (*testing.common).Logf (govet)
				t.Logf(tt.frameToSort.StringTable(-1, -1))

data/sqlutil/converter_test.go:176:14: SA1019: reflect.PtrTo has been deprecated since Go 1.22 and an alternative has been available since Go 1.18: Superseded by [PointerTo]. (staticcheck)
					kind := reflect.PtrTo(v.Type).String()
					        ^
experimental/schemabuilder/enums.go:33:29: SA1019: ast.Package has been deprecated since Go 1.22 and an alternative has been available since Go 1.0: use the type checker [go/types] instead; see [Object]. (staticcheck)
	dict := make(map[string][]*ast.Package)
```

I could not find solution for 
<pre>
printf: non-constant format string in call to (*testing.common).Logf (govet) -- <b>variables passed are the result of a fmt.Sprintf() or just strings.</b>
G115: integer overflow conversion int64 -> int32 - <b> we have multiple of these, people are disabling this due to multiple false positives. </b> 
A1019: ast.Package has been deprecated since Go 1.22 and an alternative has been available since Go 1.0: use the type checker [go/types] instead; see [Object]. (staticcheck)
</pre>